scipy/signal/_spectral_py.py

```diff
@@ -285,7 +288,8 @@ def lombscargle(
             CS -= C_sum * S_sum
 
         # determinate of the system of linear equations
-        D = CC * SS - CS * CS
+        # eps is added to prevent spurious numerical issues around the "pseudo-Nyquist"
+        D = CC * SS - CS * CS + eps
```

Comment:
Better do this only in the vicinity of 0.0 like 

```python
D = CC*SS - CS*CS
if (D < eps):
    D += eps
```

Reply:
That won't work, as sometimes (I can see from the tests I've run), D is negative.

Reply:
You can add an abs. Otherwise you can zero out an nonzero D.

Reply:
Just `max(CC*SS-CS*CS, eps)`?

Reply:
> You can add an abs. Otherwise you can zero out an nonzero D. The point only do this when D is close to the risky case.

@ilayn For some reason you're looking at the older commit (before this comment thread). I removed that code nearly immediately and replaced it with a conditional test.

> just `max(CC*SS-CS*CS, eps)`?

@mhvk Same problem as above. I've seen that sometimes D is negative, so that sign needs to propagate. The only actual issue is `D==0`. D very very very close to zero is perfectly fine (the numerator cancels out what otherwise would be a huge value).